validator_manager/src/list_validators.rs

```diff
@@ -0,0 +1,206 @@
+use std::path::PathBuf;
+
+use clap::{Arg, ArgAction, ArgMatches, Command};
+use eth2::SensitiveUrl;
+use serde::{Deserialize, Serialize};
+
+use crate::{common::vc_http_client, DumpConfig};
+
+pub const CMD: &str = "list";
+pub const VC_URL_FLAG: &str = "vc-url";
+pub const VC_TOKEN_FLAG: &str = "vc-token";
+
+pub fn cli_app() -> Command {
+    Command::new(CMD)
+        .about("Lists all validators in a validator client using the HTTP API.")
+        .arg(
+            Arg::new(VC_URL_FLAG)
+                .long(VC_URL_FLAG)
+                .value_name("HTTP_ADDRESS")
+                .help(
+                    "A HTTP(S) address of a validator client using the keymanager-API. \
+            If this value is not supplied then a 'dry run' will be conducted where \
+            no changes are made to the validator client.",
+                )
+                .default_value("http://localhost:5062")
+                .requires(VC_TOKEN_FLAG)
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+        .arg(
+            Arg::new(VC_TOKEN_FLAG)
+                .long(VC_TOKEN_FLAG)
+                .value_name("PATH")
+                .help("The file containing a token required by the validator client.")
+                .action(ArgAction::Set)
+                .display_order(0),
+        )
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct ListConfig {
+    pub vc_url: SensitiveUrl,
+    pub vc_token_path: PathBuf,
+}
+
+impl ListConfig {
+    fn from_cli(matches: &ArgMatches) -> Result<Self, String> {
+        Ok(Self {
+            vc_token_path: clap_utils::parse_required(matches, VC_TOKEN_FLAG)?,
+            vc_url: clap_utils::parse_required(matches, VC_URL_FLAG)?,
+        })
+    }
+}
+
+pub async fn cli_run(matches: &ArgMatches, dump_config: DumpConfig) -> Result<(), String> {
+    let config = ListConfig::from_cli(matches)?;
+    if dump_config.should_exit_early(&config)? {
+        Ok(())
+    } else {
+        println!("{}", run(config).await?);
+        Ok(())
+    }
+}
+
+pub async fn run<'a>(config: ListConfig) -> Result<String, String> {
+    let ListConfig {
+        vc_url,
+        vc_token_path,
+    } = config;
+
+    let (http_client, _keystores) = vc_http_client(vc_url.clone(), &vc_token_path).await?;
```

Comment:
I think you can actually use the keystores directly here without needing to make a second call to the VC. Something like this:

```suggestion
    let (_, validators) = vc_http_client(vc_url.clone(), &vc_token_path).await?;
```